lib/caching.js

```diff
@@ -130,20 +156,120 @@ var caching = function(args) {
         });
     };
 
+    function wrapMultiple(keys, work, options, cb) {
+        /**
+         * We create a unique key for the multiple keys
+         * by concatenating them
+         */
+        var uuidKey = keys.reduce(function(acc, k) {
```

Comment:
Maybe call this `combinedKey` or something... `uuid` implies that it's and actual `uuid` (per RFC spec, etc.)

Reply:
Ok, I will change it.